pkg/registry/core/service/rest.go

```diff
@@ -507,6 +517,17 @@ func contains(haystack []int, needle int) bool {
 	return false
 }
 
+// This is O(N), but we expect serviceNodePorts to be small;
+// so small that we expect a linear search to be faster
+func containsNodePorts(serviceNodePorts []ServiceNodePort, serviceNodePort ServiceNodePort) bool {
+	for _, snp := range serviceNodePorts {
+		if snp.Protocol == serviceNodePort.Protocol && snp.NodePort == serviceNodePort.NodePort {
```

Comment:
Since here both `snp` and `serviceNodePort` are instances of type `ServiceNodePort`, I suggest comparing them using `reflect.DeepEqual`.